### PR TITLE
Question and comment

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added the following features:
- Add, Edit, and Delete questions
- Add comment to posts and questions

### Files to review:
- components/card/QuestionCard.jsx
- components/comment/Comment.jsx
- features/home/questionSlice.js
- features/singlePost
- features/singleQuestion
- services/questionServices
- services/commentServices

### Files to ignore
- All files with minor modifications
- Controllers

### CODE REVIEW

File: src/components/card/QuestionCard.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        <h3 className="text-xl font-semibold hover:underline">
+          {question.questionTitle}
+        </h3>
+        <p className="pt-2 text-gray-600 whitespace-pre-wrap">
+          {question.questionContent}
+        </p>
+      </section>
+    </article>
+  ) : (
+    <></>
```
Comment (mikesch803) [inline src/components/card/QuestionCard.jsx:82]: why this empty fragment?

**Review by mikesch803 — APPROVED** [review]
looks good to me, nothing to suggest.

**Review by Vishnu-Aithal — APPROVED** [review]
Code is Good And Readable, nice UI, all functionalities are working, Good to Merge.
Suggestions - new Question can be added to top, now its added at bottom.
try wrapping all network calls with try catch

File: src/components/card/QuestionCard.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/components/card/QuestionCard.jsx:82]: I am displaying an empty fragment just in case I don't find the user that posted a question.

**Comment by ApurvaSawant11** [comment]
> Code is Good And Readable, nice UI, all functionalities are working, Good to Merge. Suggestions - new Question can be added to top, now its added at bottom. try wrapping all network calls with try catch

Thanks for the review,
1. Yes, I will add new posts/questions on the top when I will implement the sort functionality.
2. I have wrapped all the network calls in the try-catch block. Can you please let me know where I have missed it?

**Comment by Vishnu-Aithal** [comment]
> > Code is Good And Readable, nice UI, all functionalities are working, Good to Merge. Suggestions - new Question can be added to top, now its added at bottom. try wrapping all network calls with try catch
> 
> Thanks for the review,
> 
> 1. Yes, I will add new posts/questions on the top when I will implement the sort functionality.
> 2. I have wrapped all the network calls in the try-catch block. Can you please let me know where I have missed it?

Sorry! MyBad I was just looking in the Services file, its wrapped in reducer,

**Pull request merged**